v2/pkg/engine/resolve/loader.go

```diff
@@ -813,6 +816,29 @@ func (l *Loader) optionallyOmitErrorExtensions(values []*astjson.Value) {
 	}
 }
 
+// optionallyOmitCustomErrorFields removes all custom fields from the subgraph error
+func (l *Loader) optionallyOmitCustomErrorFields(values []*astjson.Value) {
+	if !l.omitCustomSubgraphErrorFields {
+		return
+	}
+	for _, value := range values {
+		if value.Type() == astjson.TypeObject {
+			obj := value.GetObject()
+			var keysToDelete []string
+			obj.Visit(func(k []byte, v *astjson.Value) {
+				key := unsafebytes.BytesToString(k)
+				if key != "message" && key != "extensions" {
```

Comment:
what if
```json
{"extensions":{"foo":"bar"}}
```

Reply:
Wouldn't it make more sense to explicitly whitelist?

Reply:
This is handled with `allowed_extension_fields`